### DISCUSSION TITLE
Fresh project build errors - ESP Insights Project commit: HEAD-HASH-NOTFOUND & ESP RainMaker Project commit: HEAD-HASH-NOTFOUND (IDFGH-14314)

### DISCUSSION THREAD

**Discussion by ae-5th** [espressif/arduino-esp32#11119] [Question - Community Help]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v5.3.2

### Operating System used.

Windows

### How did you build your project?

VS Code IDE

### If you are using Windows, please specify command line type.

None

### What is the expected behavior?

Build without errors.

### What is the actual behavior?

Build with errors.

### Steps to reproduce.

Go to New Project Wizard - Choose Template arduino-as-component.
Then CTRL+Shift+P and type/choose ESP-IDF: Add Arduino ESP32 as ESP-IDF Component.

### Build or installation Logs.

```plain
-- git rev-parse returned 'fatal: not a git repository (or any of the parent directories): .git'
ESP Insights Project commit: HEAD-HASH-NOTFOUND
-- git rev-parse returned 'fatal: not a git repository (or any of the parent directories): .git'
ESP RainMaker Project commit: HEAD-HASH-NOTFOUND
CMake Error at components/arduino/CMakeLists.txt:362 (message):
  esp32-arduino requires CONFIG_FREERTOS_HZ=1000 (currently 100)
```

### More Information.

Fresh project with Arduino component does not build.

**Comment by me-no-dev** [comment]
The actual issue is `esp32-arduino requires CONFIG_FREERTOS_HZ=1000 (currently 100)`.
Try adding `CONFIG_FREERTOS_HZ=1000` to `sdkconfig.defaults` inside the project (create the file if not there) and try to build again

**Comment by ae-5th** [comment]
That did not helped. Still same build errors.

**Comment by ResRipper** [comment]
They are two separate issues:
1. esp32-arduino requires CONFIG_FREERTOS_HZ=1000 (currently 100)
    - Set `CONFIG_FREERTOS_HZ=1000` in `sdkconfig`
2. Project commit: HEAD-HASH-NOTFOUND (https://github.com/espressif/esp-matter/issues/881 https://github.com/espressif/esp-idf/issues/7151)
    - Go to `build/esp-idf/espressif__esp_rainmaker/CMakeFiles/git-data` and `build/esp-idf/espressif__esp_insights/CMakeFiles/git-data`
    - Rename `HEAD` file to `head-ref`
    - Rerun the reconfigure task (Press `F1`, find the `ESP-IDF: Run idf.py reconfigure Task`)

**Reply by geiseri** [reply]
Is there a more automated way to do this?  The rename that is.  Currently I just use a shell script so its no biggie.
